packages/react/src/primitives/shared/styleUtils.ts

```diff
@@ -0,0 +1,167 @@
+      rowFromSpanValue: convertGridSpan(rowSpan) as GridItemStyleProps['row'],
+      columnFromSpanValue: convertGridSpan(
+        columnSpan
+      ) as GridItemStyleProps['column'],
+    };
+  }, [rowSpan, columnSpan, convertGridSpan]);
+
+  return {
+    row: !isNullOrEmptyString(row) ? row : rowFromSpanValue,
+    column: !isNullOrEmptyString(column) ? column : columnFromSpanValue,
+    ...rest,
+  };
+};
+
+export const usePropStyles = (props: ViewProps, style: any) => {
+  const {
+    theme: {
+      breakpoints: {
+        values: breakpoints,
+        unit: breakpointUnit,
+        defaultBreakpoint,
+      },
+    },
+  } = useTheming();
+
+  const breakpoint = useBreakpoint({
+    breakpoints,
+    breakpointUnit,
+    defaultBreakpoint: defaultBreakpoint as Breakpoint,
+  });
+
+  const propStyles = useTransformStyleProps(props);
+
+  return React.useMemo(
+    () =>
+      prefixer(
+        convertStylePropsToStyleObj({
+          props: propStyles,
+          style,
+          breakpoint,
+          breakpoints,
+        })
+      ),
+    [props, style, breakpoints, breakpoint]
+  );
+};
+
+interface convertStylePropsToStyleObjParams {
+  props: ViewProps;
+  style?: React.CSSProperties;
+  breakpoint: Breakpoint;
+  breakpoints: Breakpoints;
+}
+export interface ConvertStylePropsToStyleObj {
+  (params: convertStylePropsToStyleObjParams);
+}
+
+export const isSpanPrimitiveValue = (
+  spanValue: GridItemStyleProps['rowSpan'] | GridItemStyleProps['columnSpan']
+): spanValue is 'auto' | number => {
+  return (
+    spanValue === 'auto' || (typeof spanValue === 'number' && !isNaN(spanValue))
+  );
+};
+
+export const convertGridSpan = (
+  spanValue: GridItemStyleProps['rowSpan'] | GridItemStyleProps['columnSpan']
+): GridItemStyleProps['row'] | GridItemStyleProps['column'] => {
+  if (isSpanPrimitiveValue(spanValue)) {
+    return getGridSpan(spanValue);
+  }
+  if (typeof spanValue === 'object' && spanValue != null) {
+    if (Array.isArray(spanValue)) {
+      return spanValue.map((value) => getGridSpan(value));
+    } else {
+      const newObj: ResponsiveObject<string> = {};
+      Object.entries(spanValue).forEach(([key, value]) => {
+        newObj[key] = getGridSpan(value);
+      });
+      return newObj;
+    }
+  }
+  return null;
+};
+
+export const getGridSpan = (spanValue: number | 'auto'): string => {
+  return spanValue === 'auto'
+    ? 'auto'
+    : `span ${spanValue} / span ${spanValue}`;
+};
+
+/**
+ * Convert style props to CSS variables for React style prop
+ * Note: Will filter out undefined, null, and empty string prop values
+ * @returns CSSProperties styles
+ */
+export const convertStylePropsToStyleObj: ConvertStylePropsToStyleObj = ({
+  props = {},
+  style = {},
+  breakpoint,
+  breakpoints,
+}) => {
+  ComponentPropsToStylePropsMapKeys.filter((stylePropKey) =>
+    filterOutNullOrEmptyStringValues(props[stylePropKey])
+  ).forEach((stylePropKey) => {
+    let value = getValueAtCurrentBreakpoint(
```

Comment:
if `value` never change, can be a `const`?

Reply:
Good catch, will update